doc/api/worker.md

```diff
@@ -0,0 +1,255 @@
+# Worker
+
+> Stability: 1 - Experimental
+
+The `worker` module provides a way to create multiple environments running
+on independent threads, and to create message channels between those. It
+can be accessed using:
+
+```js
+const worker = require('worker');
+```
+
+Workers are useful for performing CPU-intensive JavaScript operations; do not
+use them for I/O, since Node’s built-in mechanisms for performing operations
+asynchronously already treat it more efficiently than Worker threads can.
+
+## Example
+
+```js
+const { Worker } = require('worker');
+
+if (process.isMainThread) {
+  module.exports = async function parseJSAsync(script) {
+    return new Promise((resolve, reject) => {
+      const worker = new Worker(__filename, {
+        workerData: script
+      });
+      worker.on('message', resolve);
+      worker.on('error', reject);
+      worker.on('exit', (code) => {
+        if (code !== 0)
+          reject(new Error(`Worker stopped with exit code ${code}`));
+      });
+    });
+  };
+} else {
+  const { parse } = require('some-js-parsing-library');
+  const script = process.workerData;
+  process.postMessage(parse(script));
+}
+```
+
+## Class: MessageChannel
+<!-- YAML
+added: REPLACEME
+-->
+
+The `MessageChannel` has no methods of its own. `new MessageChannel()`
```

Comment:
Are these compatible with DOM message channels? If so we should link to the DOM spec on those here somewhere.

Reply:
Almost – this is doing `MessageChannel`s as Node `EventEmitter`s rather than `EventTarget`s, but otherwise it should work the same